src/Components/LetterBuilder/FormSteps/PreviewStep.tsx

```diff
@@ -1,17 +1,75 @@
+import { useEffect, useState } from "react";
+import { useLingui } from "@lingui/react";
 import { Trans } from "@lingui/react/macro";
+import { msg } from "@lingui/core/macro";
 
 import { FormHookProps } from "../../../types/LetterFormTypes";
-import { LetterPreview } from "../Letter/Letter";
+import { languageNames, SupportedLocale } from "../../../i18n-base";
+import { buildLetterHtml } from "../Letter/letter-utils";
 
 export const PreviewStep: React.FC<FormHookProps> = (props) => {
   const { getValues } = props;
 
+  const { i18n, _ } = useLingui();
+  const letterData = getValues();
+
+  const [letter, setLetter] = useState<string | undefined>();
+  const [previewLocale, setPreviewLocale] = useState<SupportedLocale>(
+    i18n.locale as SupportedLocale
+  );
+
+  useEffect(() => {
+    const prepLetter = async () => {
+      const localizedLetterHtml = await buildLetterHtml(
+        letterData,
+        previewLocale
+      );
+      setLetter(localizedLetterHtml);
+    };
+    prepLetter();
+  }, [letterData, previewLocale]);
```

Comment:
```suggestion
  }, [previewLocale]);
```

don't think letterData is necessary here since we shouldn't expect it to change on this page